src/PyBytesProxyHandler.cc

```diff
@@ -39,22 +39,26 @@ static bool array_valueOf(JSContext *cx, unsigned argc, JS::Value *vp) {
   for (size_t i = 0; i < byteLength; i++) {
     numberOfDigits += data[i] < 10 ? 1 : data[i] < 100 ? 2 : 3;
   }
+
   const size_t STRING_LENGTH = byteLength + numberOfDigits;
   JS::Latin1Char *buffer = (JS::Latin1Char *)malloc(sizeof(JS::Latin1Char) * STRING_LENGTH);
 
-  size_t charIndex = 0;
-  snprintf((char *)&buffer[charIndex], 4, "%d", data[0]);
-  charIndex += data[0] < 10 ? 1 : data[0] < 100 ? 2 : 3;
+  if (snprintf((char *)&buffer[0], 3 + 1, "%hu", data[0]) < 0) {
```

Comment:
Curious - what is the intention of the < 0 test?  I don't think snprintf can actually return -1 with a valid format string except in the case where it can't allocate memory on the stack ... and in that case, error recovery is basically hopeless.

Do we care that malloc could return NULL here? 

I like how you wrote `3 + 1` instead of `4` here. It makes the meaning very clear.

Reply:
this is what does the trick of removing the warning